dfd-wrapper/src/test/java/com/revolut/rxdata/dfd_wrapper/DfdFunctionalTest.kt

```diff
@@ -0,0 +1,184 @@
+package com.revolut.rxdata.dfd_wrapper
+
+import app.cash.turbine.test
+import com.revolut.data.model.Data
+import com.revolut.flowdata.extensions.takeUntilLoaded
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+import org.junit.jupiter.api.RepeatedTest
+import org.junit.jupiter.api.RepetitionInfo
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS
+import org.junit.jupiter.api.parallel.Execution
+import org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.random.Random
+import kotlin.time.Duration.Companion.milliseconds
+
+/*
+ * Copyright (C) 2023 Revolut
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+@ExperimentalCoroutinesApi
+@TestInstance(PER_CLASS)
+class DfdFunctionalTest {
```

Comment:
also don't think this should be retested with the new contract. switching api from rx to coroutines shouldn't introduce new internal race issues to DOD; but if it does than only those specific cases should be left here

Reply:
Done 😁